src/main.rs

```diff
@@ -268,3 +275,324 @@ async fn main() {
+        }
+        unpack(&env.shasum());
+
+        // Assert the files and directory we expect to exist do exist
+        assert!(Path::new(&output_root.join("index.json")).exists());
+        assert!(Path::new(&output_root.join("manifest.json")).exists());
+        assert!(Path::new(&output_root.join("oci-layout")).exists());
+        assert!(Path::new(&output_root.join("repositories")).exists());
+
+        localize_test_image(TEST_IMAGE, &output_root)
+            .expect("should have localized the test image's index.json");
+
+        // Use :0 to let the operating system decide the random port to listen on
+        let listener = tokio::net::TcpListener::bind("127.0.0.1:0")
+            .await
+            .expect("should have been able to bind listener to a random port on localhost");
+        let random_port = listener
+            .local_addr()
+            .expect("should have been able to resolve the address")
+            .port();
+
+        // Start registry in the background
+        tokio::spawn(async {
+            let app = start_seed_registry();
+            axum::serve(listener, app)
+                .await
+                .expect("should have been able to start serving the registry");
+        });
+
+        let test_image = TEST_IMAGE.replace("ghcr.io", &format!("127.0.0.1:{random_port}"));
+        let options = Some(CreateImageOptions {
+            from_image: test_image.clone(),
+            ..Default::default()
+        });
+
+        let test_image_pull = docker
+            .create_image(options, None, None)
+            .try_collect::<Vec<_>>()
+            .await;
+        assert!(test_image_pull.is_ok());
+        docker
+            .remove_image(&test_image, None, None)
+            .await
+            .expect("should have cleaned up the pulled test image");
+    }
+
+    // This localizes the test image's index.json such that the registry server
+    // will be able to match the test image from it
+    fn localize_test_image(image_reference: &str, image_root: &Path) -> Result<()> {
+        let reference: String = normalize_manifest_reference(image_reference)
+            .context("should have localized the test image reference")?;
+
+        let mut index_file = File::options()
+            .read(true)
+            .write(true)
+            .open(image_root.join("index.json"))
+            .context("should have opened index.json")?;
+
+        let mut index_json: serde_json::Value =
+            serde_json::from_reader(index_file.try_clone().unwrap())
+                .context("should have read index.json")?;
+
+        // Overwrite or add an annotation for "org.opencontainers.image.base.name"
+        // that is normalized to be without registry address so that it can be
+        // pulled locally
+        index_json
+            .get_mut("manifests")
+            .and_then(|manifests| manifests.get_mut(0))
+            .and_then(|array| array.get_mut("annotations"))
```

Comment:
if annotations is empty - does this work as intended? I recall similar problems throughout zarf with regards to empty and wanted to verify.